src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -1697,14 +1697,38 @@ bool LibraryCallKit::inline_vector_compare() {
   Node* v1 = unbox_vector(argument(5), vbox_type, elem_bt, num_elem);
   Node* v2 = unbox_vector(argument(6), vbox_type, elem_bt, num_elem);
 
+  bool is_masked_op = argument(7)->bottom_type() != TypePtr::NULL_PTR;
+  Node* mask = is_masked_op ? unbox_vector(argument(7), mbox_type, elem_bt, num_elem) : NULL;
```

Comment:
It's better to check whether the` "unbox_vector"` result for `mask` is ` "null"` if "`is_masked_op`" is true. And return false if it is.